File: 1300.cc
#include <bits/stdc++.h>
using namespace std;

long long n,k;
int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    cin >> n;
    cin >> k;

    int start = 1; int end = k;
    long long result = 0;
    
    while(start <= end){
        long long mid = (start + end)/2;

        long long count = 0;
        for(int i = 1; i <= n; i++){
            count += min(n,mid/i);
        }

        if(count < k){
            start = mid + 1;
        }else{
            result = mid;
            end = mid-1;
        }
    }

    cout << result << endl; 
}

File: Python/1300.py
n = int(input())
k = int(input())

start = 1
end = k

while start <= end:
    mid = (start + end)//2
    count = 0
    for i in range(1,n+1):
        count += min(mid//i, n)

    if count < k:
        start = mid + 1

    else:
        end = mid - 1

print(start)